### DISCUSSION TITLE
Was there ever a version available for Android 4.0 or could I easily modify an existing version to work on Android 4.0?

### DISCUSSION THREAD

**Discussion by hamishmb** [streetcomplete/StreetComplete#4073] [Q&A] [answered]
Was there ever a version available for Android 4.0 or could I easily modify an existing version to work on Android 4.0?

**Comment by matkoniecz** [comment, marked as answer]
https://github.com/streetcomplete/StreetComplete/blob/ff7d70d903e3a7b65964d7dc9daa4b716461f795/app/src/main/res/raw/changelog.yml#L345

v34.2 was last version with Android 4.2 support, v35.0 has "Android 4.x support has been dropped"

If keeping support would be easy then it would be kept. https://github.com/streetcomplete/StreetComplete/releases/tag/v34.2 has file that likely can be still installed (after enabling .apk file use on Android - note that there are some attacks that start from installing .apk file with malware! )

https://github.com/streetcomplete/StreetComplete/releases has all releases and https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/res/raw/changelog.yml entire changelog

EDIT: misleading info removed, see below

**Reply by Helium314** [reply]
Actually Android 4.0 is API level 14, which was never supported (looking through build.gradle in git history).
The lowest version used in SC was Android 4.2 (API 17)

**Reply by hamishmb** [reply]
Maybe "easy" was the wrong word - I know it's not easy :)

I think I may have a better and more fulfilling time porting a newer version of Android to my old 4.0 device rather than trying to shoehorn an older version of StreetComplete onto it. Thanks both :)

**Reply by hamishmb** [reply]
Unsurprisingly, my attempt to modify the mid sdk version to 15 down from 17 just results in an app that crashes after a long load time. I wanted a newer android version on this old phone anyway so that's fine. I may experiment with this more at some point, but no need just as of yet. Apparently only one component is incompatible though, so it might not be too much of a stretch to replace/downgrade it.

**Reply by hamishmb** [reply]
So, I got Android 4.4 on the device, but I get no map and scanning for quests instantly fails. I guess maybe 256MB (or possibly 384MB, unclear) of RAM is simple not enough?

**Reply by Helium314** [reply]
Maybe if you have root access you could increase `dalvik.vm.heapgrowthlimit`, see #3656 (though 192 might be too much for your phone zu handle).

**Reply by hamishmb** [reply]
Hmm, interesting. I'll give it a go. It turns out I don't need this to work any more, but it's still an interesting thought experiment to see how far I can take it.
